icicle/appUtils/msm/commit.cu

```diff
@@ -0,0 +1,57 @@
+#include <cuda.h>
+#include "commit.cuh"
+#include "msm.cuh"
+
+
+/**
+ * Commit to a polynomial using the MSM.
+ * Note: this function just calls the MSM, it doesn't convert between evaluation and coefficient form of scalars or points.
+ * @param d_out Ouptut point to write the result to.
+ * @param d_scalars Scalars for the MSM. Must be on device.
+ * @param d_points Points for the MSM. Must be on device.
+ * @param count Length of `d_scalars` and `d_points` arrays (they should have equal length).
+ */
+extern "C"
+projective_t* commit_cuda(scalar_t* d_scalars, affine_t* d_points, size_t count, size_t device_id = 0)
+{
+    try
+    {
+        projective_t* d_out;
+        cudaMalloc(&d_out, sizeof(projective_t));
+        // TODO: set c depending on `count` instead of just 10
+        bucket_method_msm(scalar_t::NBITS, 10, d_scalars, d_points, count, d_out, true);
```

Comment:
the wrapper "large_msm" should be used instead of directly using "bucket_methd_msm" and the value of c needs to be determined there